src/gpg_ux_nbgl.c

```diff
@@ -0,0 +1,1116 @@
+            attributes.length = 1 + oid_len;
+            break;
+    }
+
+    switch (key_name) {
+        case TOKEN_TEMPLATE_SIG:
+            dest = &G_gpg_vstate.kslot->sig;
+            break;
+        case TOKEN_TEMPLATE_DEC:
+            dest = &G_gpg_vstate.kslot->dec;
+            break;
+        case TOKEN_TEMPLATE_AUT:
+            dest = &G_gpg_vstate.kslot->aut;
+            break;
+    }
+
+    if (dest && attributes.value[0]) {
+        nvm_write(dest, NULL, sizeof(gpg_key_t));
+        nvm_write(&dest->attributes, &attributes, sizeof(attributes));
+    }
+    ui_menu_settings();
+}
+
+static bool key_type_nav_cb(uint8_t page, nbgl_pageContent_t* content) {
+    UNUSED(page);
+    static const uint8_t barTokens[] = {TOKEN_TYPE_RSA2048,
+                                        TOKEN_TYPE_RSA3072,
+                                        TOKEN_TYPE_RSA4096,
+                                        TOKEN_TYPE_SECP256K1,
+                                        TOKEN_TYPE_Ed25519};
+
+    memset(content, 0, sizeof(nbgl_pageContent_t));
+    content->type = BARS_LIST;
+    content->barsList.barTexts = keyTypeTexts;
+    content->barsList.tokens = barTokens;
+    content->barsList.nbBars = ARRAYLEN(barTokens);
+    content->barsList.tuneId = TUNE_TAP_CASUAL;
+    return true;
+}
+
+static void key_type_ctrl_cb(int token, uint8_t index) {
+    UNUSED(index);
+
+    G_gpg_vstate.ux_type = token - FIRST_USER_TOKEN;
+
+    snprintf(G_gpg_vstate.menu,
+             sizeof(G_gpg_vstate.menu),
+             "%s - %s",
+             (char*) PIC(keyNameTexts[G_gpg_vstate.ux_key]),
+             (char*) PIC(keyTypeTexts[G_gpg_vstate.ux_type]));
+    nbgl_useCaseConfirm("Set Template", G_gpg_vstate.menu, "Confirm", "Cancel", key_confirm_cb);
+}
+
+static bool template_nav_cb(uint8_t page, nbgl_pageContent_t* content) {
+    UNUSED(page);
+    static const uint8_t barTokens[] = {TOKEN_TEMPLATE_SIG, TOKEN_TEMPLATE_DEC, TOKEN_TEMPLATE_AUT};
+    static char* names[KEY_NB] = {0};
+    static char text[KEY_NB][64];
+    u_int8_t key;
+
+    memset(content, 0, sizeof(nbgl_pageContent_t));
+    for (key = 0; key < KEY_NB; key++) {
+        snprintf(text[key],
+                 sizeof(text[key]),
+                 "%s [%s]",
+                 (char*) PIC(keyNameTexts[key]),
+                 _getKeyType(barTokens[key]));
+        names[key] = text[key];
+    }
+
+    content->type = BARS_LIST;
+    content->barsList.barTexts = (const char* const*) names;
+    content->barsList.tokens = barTokens;
+    content->barsList.nbBars = ARRAYLEN(barTokens);
+    content->barsList.tuneId = TUNE_TAP_CASUAL;
+    return true;
+}
+
+static void template_ctrl_cb(int token, uint8_t index) {
+    UNUSED(index);
+    if ((token < TOKEN_TEMPLATE_SIG) || (token > TOKEN_TEMPLATE_AUT)) {
+        return;
+    }
+    G_gpg_vstate.ux_key = token - FIRST_USER_TOKEN;
+    nbgl_useCaseSettings((char*) PIC(keyNameTexts[G_gpg_vstate.ux_key]),
```

Comment:
```suggestion
    nbgl_useCaseSettings((const char*) PIC(keyNameTexts[G_gpg_vstate.ux_key]),
```

Reply:
Can be done on a bunch of other PIC in the file

Reply:
Fixed